projects/js-packages/connection/state/actions.jsx

```diff
@@ -20,6 +35,34 @@ const connectionStatusActions = {
 	setUserIsConnecting: isConnecting => {
 		return { type: SET_USER_IS_CONNECTING, isConnecting };
 	},
+	setRegistrationError: registrationError => {
+		return { type: SET_REGISTRATION_ERROR, registrationError };
+	},
+	clearRegistrationError: () => {
+		return { type: CLEAR_REGISTRATION_ERROR };
+	},
+	setAuthorizationUrl: authorizationUrl => {
+		return { type: SET_AUTHORIZATION_URL, authorizationUrl };
+	},
+	*registerSite( registrationNonce, redirectUri ) {
+		yield dispatch( STORE_ID ).clearRegistrationError();
+		yield dispatch( STORE_ID ).setSiteIsRegistering( true );
+		yield restApi
+			.registerSite( registrationNonce, redirectUri )
+			.then( response => {
+				dispatch( STORE_ID ).setSiteIsRegistering( false );
+				dispatch( STORE_ID ).setConnectionStatus( { isRegistered: true } );
+				dispatch( STORE_ID ).setAuthorizationUrl( response.authorizeUrl );
+				dispatch( STORE_ID ).setUserIsConnecting( true );
```

Comment:
We might consider having the `register` (site connection) functionality independent of the user connection. I guess that would mean moving this to `ConnectButton` or `ConnectUser`?

Reply:
Can you expand on that?

This is related to my first comment above, about how to hook to the end of an action.

I think we can get rid of the `ConnectUser` component. It was useful when we had both the redirect and the iframe possibilities. Now we are always redirecting... I would rename this `isUserConnecting` var to something like `redirectToAuthorizationNow`...

Reply:
Mainly thinking about cases where consumers, like Boost, are interested in establishing a site only connection. 
I think ideally the consumer should inform us about it and then we would only redirect if the consumer requires a full connection.

> I think we can get rid of the ConnectUser component.

Yes, I agree. The whole logic can be moved to state instead.

Reply:
> The whole logic can be moved to state instead.

I thought this at first, but I was unsure about who should be responsible for the redirect. Certainly not the store. So something has to stay in the ConnectButton... That's why I decided to not change this on this PR. Let's handle this in separate PR